lib/aempartsize.js

```diff
@@ -40,15 +41,15 @@ function clamp(value, min, max) {
  */
 function calculatePartSize(numUrls, fileSize, minPartSize, maxPartSize, preferredPartSize) {
     if (numUrls <= 0) {
-        throw Error(`Invalid number of urls: ${numUrls}`);
+        throw new IllegalArgumentError("Invalid number of urls", numUrls);
     } else if (fileSize <= 0) {
-        throw Error(`Unsupported file size: ${fileSize}`);
+        throw new IllegalArgumentError("Unsupported file size", fileSize);
```

Comment:
For this one it would be good to elaborate the error text - because from the text you can't know if the issue is that the file is too large, too small (?!?) or the value is not set...